object-filter/src/main/java/org/infinispan/objectfilter/impl/hql/FilterRendererDelegate.java

```diff
@@ -20,31 +22,80 @@
 
    private final ObjectPropertyHelper<TypeMetadata> propertyHelper;
 
+   private final SingleEntityHavingQueryBuilder<BooleanExpr> havingBuilder;
+
    private TypeMetadata targetEntityMetadata;
 
+   private List<PropertyPath> groupBy;
+
    private List<SortField> sortFields;
 
+   private final List<PropertyPath> projections = new ArrayList<PropertyPath>();
+
    public FilterRendererDelegate(EntityNamesResolver entityNamesResolver, ObjectPropertyHelper<TypeMetadata> propertyHelper,
-                                 SingleEntityQueryBuilder<BooleanExpr> builder, Map<String, Object> namedParameters) {
+                                 SingleEntityQueryBuilder<BooleanExpr> builder, SingleEntityHavingQueryBuilder<BooleanExpr> havingBuilder, Map<String, Object> namedParameters) {
       super(propertyHelper, entityNamesResolver, builder, namedParameters);
       this.propertyHelper = propertyHelper;
+      this.havingBuilder = havingBuilder;
    }
 
    @Override
-   protected void addSortField(PropertyPath propertyPath, String collateName, boolean isAscending) {
+   protected SingleEntityHavingQueryBuilder<BooleanExpr> getHavingBuilder() {
+      return havingBuilder;
+   }
+
+   @Override
+   protected void addSortField(org.hibernate.hql.ast.origin.hql.resolve.path.PropertyPath propertyPath, String collateName, boolean isAscending) {
+      // collationName is ignored
       if (sortFields == null) {
-         sortFields = new ArrayList<SortField>();
+         sortFields = new ArrayList<SortField>(5);
+      }
+      sortFields.add(new FilterParsingResult.SortFieldImpl(makePropertyPath(propertyPath), isAscending));
+   }
+
+   private PropertyPath makePropertyPath(org.hibernate.hql.ast.origin.hql.resolve.path.PropertyPath propertyPath) {
+      AggregationPropertyPath.Type aggregationType = propertyPath instanceof AggregationPropertyPath ? ((AggregationPropertyPath) propertyPath).getType() : null;
+      PropertyPath.AggregationType _aggregationType = null;
+      if (aggregationType != null) {
```

Comment:
You already have this code https://github.com/infinispan/infinispan/pull/3591/files#diff-0bccd6825414ee3a8e5e98e6e38c4688R143
Put it somewhere reusable